### ISSUE TITLE
Raptor Cybernetics just break randomly and turn into the humanoid version on Teshari

### ISSUE THREAD

**Issue by Redrover1760** [Bubberstation/Bubberstation#1016]
## Issue Summary

My precious friend Duey has a fully prosthetic teshari with all limbs, chest, and head replaced with the robotic raptor and a kitsune tail. Very, very commonly, out of nowhere the limbs break and default to the the humanoid robotic limb sprites. In fact, I amputated them, they became normal robotic limbs instead of raptor ones, replacing did fix it until it broke again, however.

Its just crazy and my friend is really bummed out about this. I'm willing to offer up to 5 dollars to anyone who fixes this if they are interested.

Also, robotic eyes do not respect eye color, they force white eyes onto teshari, at the very least. If you could make it so they do, that'd be great.

## Round ID:

3315, Many others

## Testmerges:

[#935: 'bubber always message staffchat ahelps' by nevimer at commit 4c8172eb1c](https://github.com/Bubberstation/Bubberstation/pull/935)
[#910: 'Nerfs most modsuits by giving them all armour boosters' by YakumoChen at commit 72f346e697](https://github.com/Bubberstation/Bubberstation/pull/910)
[#965: '[Early Mirror] Server announcements are no longer ock ick' by lessthnthree at commit e5cc02e2d0](https://github.com/Bubberstation/Bubberstation/pull/965)
[#968: '[Early Mirror] Tarkon Retarken, The great Atarkoining' by Erol509 at commit e91f203939](https://github.com/Bubberstation/Bubberstation/pull/968)
[#625: 'Contractor and Ninja Modsuits have a deployable microbomb (Enable the DNA lock and hit Lock on the modsuit to arm)' by nevimer at commit 1a9af007c8](https://github.com/Bubberstation/Bubberstation/pull/625)
[#987: 'Ports Bloopers (Vocal Barks) from FluffySTG' by shayoki at commit a645f6e07f](https://github.com/Bubberstation/Bubberstation/pull/987)
[#798: '[Will be MODULAR] Adds back TEG ' by Erol509 at commit 543d180fdf](https://github.com/Bubberstation/Bubberstation/pull/798)
[#962: 'The Tajara DLC! Finishes Tajara up to be an actual species now.' by pixelkitty286 at commit 512d55133f](https://github.com/Bubberstation/Bubberstation/pull/962)
[#990: '[Do not merge] removes headsets from ghost cafe' by StrangeWeirdKitten at commit 121b897f44](https://github.com/Bubberstation/Bubberstation/pull/990)
[#921: 'Adds Moon Station: AKA Burgerstation 2' by BurgerLUA at commit 1dc19fb838](https://github.com/Bubberstation/Bubberstation/pull/921)

## Reproduction:

Unsure, they seem to break at whim.

**Comment by aKromatopzia** [comment]
Right, some questions about the bug:

- Did the item description change? The raptor versions mention "membranous feathers
- Randomly as in, literally randomly? With nothing at all happening?
- How soon did this occur after the limbs were replaced, roughly?

**Comment by Redrover1760** [comment]
> Right, some questions about the bug:
> 
>     * Did the item description change? The raptor versions mention "membranous feathers"
> 
>     * Randomly as in, literally randomly? With nothing at all happening?
> 
>     * How soon did this occur after the limbs were replaced, roughly?

1. No, the item description did not change. Apologies. In fact, the item looks like normal after its removed. In fact, replacing the same prosthetic after its cut off fixes it.
2. It seems to happen with nothing happening at all, yeah. Obvious causes like damage and such breaks it.
3. Hm... When I replaced them? Maybe like 30 minutes. Friend also says its about 30 minutes on average.

**Comment by aKromatopzia** [comment]
>     1. No, the item description did not change. Apologies. In fact, the item looks like normal after its removed. In fact, replacing the same prosthetic after its cut off fixes it.
> 
>     2. It seems to happen with nothing happening at all, yeah. Obvious causes like damage and such don't do it. "Truly random." yeah
> 
>     3. Hm... When I replaced them? Maybe like 30 minutes. Friend also says its about 30 minutes on average.

1 means that it remains the same object but with icon and icon_static overridden. Now, standing in the ghost cafe for 40 mins seems to have yielded no results. What does Duey do consistently?

**Comment by Redrover1760** [comment]
![Broken](https://github.com/Bubberstation/Bubberstation/assets/39284090/b6d532c9-81d0-4391-98da-b02d37897607)

Nothing unique, as far as I'm aware. I know it breaks, it breaks quite a lot, and across multiple shifts. Something clearly triggers it, as it doesn't even break on every limb at once, sometimes it breaks on one or two or three instead of all 4.

And btw, this is dauntless. As far as I'm aware, They didn't do much of significance.

**Comment by Redrover1760** [comment]
> > ![Broken](https://private-user-images.githubusercontent.com/39284090/296972257-b6d532c9-81d0-4391-98da-b02d37897607.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTI0MjQsIm5iZiI6MTcwNTM5MjEyNCwicGF0aCI6Ii8zOTI4NDA5MC8yOTY5NzIyNTctYjZkNTMyYzktODFkMC00MzkxLTk4ZGEtYjAyZDM3ODk3NjA3LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDA4MDIwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiODc0NDUyM2JhYzliMWY4NjA1ODAwZTQ4YjNlYWYyOTFmNjJlMjc3NTNkNDQ2NTFjYWYzNGY3ZjZhYTI5ZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VxlIXB9hcl7K4yvRy7yLYoMx2sBnNdtuPC-iv9NxKkU)
> > Nothing unique, as far as I'm aware. I know it breaks, it breaks quite a lot, and across multiple shifts. Something clearly triggers it, as it doesn't even break on every limb at once, sometimes it breaks on one or two or three instead of all 4.
> > And btw, this is dauntless. As far as I'm aware, They didn't do much of significance.
> > Its hard to know when exactly it breaks too, cause its hard to tell until the clothes are off.
> 
> That image is actually quite interesting! I took a shortcut when making the advanced models, and made them "/obj/item/bodypart/leg/left/robot/teshvali_advanced" instead of "/obj/item/bodypart/leg/left/robot/advanced/teshvali". They shouldn't be inheriting the advanced model's sprites at all. I have an idea now, but don't think it's a solution. I'll get the testmerge ready.

Curious.

Also, its not just the tesh advanced prosthetics, but yeah. I just upgraded my friend to the advanced ones when replacing. Secondly, the chest breaks too, they've told me.

**Comment by Redrover1760** [comment]
I found something that sort of caused it. Removing their Kitsune Septuple tail created a ghost tail and putting it back into their groin not only removed the tail sprite but also broke all their limbs. We uh, never did this, but I guess that sort of counts?

**Comment by aKromatopzia** [comment]
Yeah my attempt to reproduce yielded little results. This will be an issue.

**Comment by Redrover1760** [comment]
Putting on a modsuit seemed to have done something once. Have put on modsuits before, it didn't break before.

**Comment by Redrover1760** [comment]
Still not fixed as of 5/20/2024.

**Comment by Majkl-J** [comment]
Is this still an issue?

**Comment by JagOfTroy** [comment]
![Screenshot 2024-12-18 092244](https://github.com/user-attachments/assets/47d712df-5dbf-4d2d-a018-c174887a39d3)
So like he said.. you can set the eye color to be whatever you want and it'll show in the loadout but once you get in-game, it defaults to the bright white.
![Screenshot 2024-12-18 092747](https://github.com/user-attachments/assets/29c1eb42-2607-46f2-bc27-533f339638d7)
After about 20 mins into the round of standing around, jumping on tables, just doing small things, I wasn't able to repro the issue.

![Screenshot 2024-12-18 092538](https://github.com/user-attachments/assets/bbf5d68b-f692-49c7-bfa6-ca386da62b3e)
![Screenshot 2024-12-18 093438](https://github.com/user-attachments/assets/4830a359-192b-4fbd-8a2d-3ab6825245c6)
![Screenshot 2024-12-18 093702](https://github.com/user-attachments/assets/38da0b4f-9bbb-4233-80f3-7bb342bb1b22)

I will note, I was using cybernetics instead of the regular prosthetics(which is what I assumed the original requester meant instead of just regular.. prosthetics for every single limb)

As far as I can tell, the issue seems resolved but will do another test run with full prosthetics instead.

![Screenshot 2024-12-18 091639](https://github.com/user-attachments/assets/4104a1a1-d370-435f-baf1-6b4122f6e6fb)
I'm also not too familiar with this race so when I saw its debuffs, I wondered if maybe something to do specific with how that race is written is what is causing the issue.

TLDR: I was not able to repro this issue but I used full raptor cybernetics, not prosthetics. The eye issue is still a thing though.

**Comment by JagOfTroy** [comment]
![Screenshot 2024-12-18 094932](https://github.com/user-attachments/assets/2ef36c66-8670-4ba4-9b38-b34aa4eb4b1b)
![Screenshot 2024-12-18 095048](https://github.com/user-attachments/assets/0d964e1b-12ea-41aa-bba9-3a55e894d942)
![Screenshot 2024-12-18 095059](https://github.com/user-attachments/assets/9e55ace8-fd1b-43a0-a918-ee98a8a0160f)


Also found weird naming conventions that don't line up in the augments menu. 
It's a prosthetic for one arm and called prototype for another? May be an overlook.

**Comment by JagOfTroy** [comment]
![Screenshot 2024-12-18 095534](https://github.com/user-attachments/assets/911a5aa0-249e-4fac-ba5a-7f7c135cd3a9)
![Screenshot 2024-12-18 100214](https://github.com/user-attachments/assets/c5ddbcbc-c01a-4071-83db-f4d28251150c)
![Screenshot 2024-12-18 100226](https://github.com/user-attachments/assets/2eb8bebe-1e2a-437e-9a46-aa3c668e2bdf)
![Screenshot 2024-12-18 103959](https://github.com/user-attachments/assets/70488082-448d-4f45-a49f-c9ddfcb1d8f7)
![Screenshot 2024-12-18 104015](https://github.com/user-attachments/assets/63eb2a63-c6c2-403f-ab4c-6c5a0d46d4cd)
These are the tests done with the prosthetic raptor limbs.

I believe this issue is fixed as I was unable to get it repro. The only thing left to fix would be the eyes if they are actually meant to change color or not for the cybernetic Raptor Head.

**Comment by Redrover1760** [comment]
I appreciate looking into this. The eyes always are white, technically replacing the eyes with cybernetic ones that changes color works to fix that.

**Comment by JagOfTroy** [comment]
> I appreciate looking into this. The eyes always are white, technically replacing the eyes with cybernetic ones that changes color works to get around the issue.

No problem! With that being said, are we still wanting to add that customization feature for the eyes? I know your able to set it and it defaults back, which is the only reason why I haven't messed with it cause either its intended that way or a bug.
